velox/functions/lib/Re2Functions.cpp

```diff
@@ -533,6 +544,114 @@ class LikeWithRe2 final : public VectorFunction {
+      VELOX_CHECK_LE(
+          compiledRegularExpressions_.size(),
+          kMaxCompiledRegexes,
+          "Max number of regex reached");
+      checkForBadPattern(*it->second);
+      return re2FullMatch(input, *it->second);
+    };
+
+    auto applyRow = [&](const StringView& input,
+                        const StringView& pattern,
+                        const std::optional<char>& escapeChar) -> bool {
+      if (!escapeChar) {
+        PatternKind patternKind;
+        vector_size_t reducedLength;
+        std::tie(patternKind, reducedLength) = determinePatternKind(pattern);
+
+        switch (patternKind) {
+          case PatternKind::kExactlyN:
+            return OptimizedLikeWithMemcmp<PatternKind::kExactlyN>::match(
+                input, pattern, reducedLength);
+          case PatternKind::kAtLeastN:
+            return OptimizedLikeWithMemcmp<PatternKind::kAtLeastN>::match(
+                input, pattern, reducedLength);
+          case PatternKind::kFixed:
+            return OptimizedLikeWithMemcmp<PatternKind::kFixed>::match(
+                input, pattern, reducedLength);
+          case PatternKind::kPrefix:
+            return OptimizedLikeWithMemcmp<PatternKind::kPrefix>::match(
+                input, pattern, reducedLength);
+          case PatternKind::kSuffix:
+            return OptimizedLikeWithMemcmp<PatternKind::kSuffix>::match(
+                input, pattern, reducedLength);
+          default:
+            return applyWithRegex(input, pattern, escapeChar);
+        }
+      }
+      return applyWithRegex(input, pattern, escapeChar);
+    };
+
+    context.ensureWritable(rows, type, localResult);
+    exec::VectorWriter<bool> vectorWriter;
+    vectorWriter.init(*localResult->asFlatVector<bool>());
+    exec::DecodedArgs decodedArgs(rows, args, context);
+
+    exec::VectorReader<Varchar> inputReader(decodedArgs.at(0));
+    exec::VectorReader<Varchar> patternReader(decodedArgs.at(1));
+
+    if (args.size() == 2) {
+      context.applyToSelectedNoThrow(rows, [&](auto row) {
+        vectorWriter.setOffset(row);
+        vectorWriter.current() =
+            applyRow(inputReader[row], patternReader[row], std::nullopt);
+        vectorWriter.commit(true);
+      });
+    } else {
+      VELOX_CHECK_EQ(args.size(), 3);
+      exec::VectorReader<Varchar> escapeReader(decodedArgs.at(2));
+      context.applyToSelectedNoThrow(rows, [&](auto row) {
+        vectorWriter.setOffset(row);
+        auto escapeChar = escapeReader[row];
+        VELOX_USER_CHECK_EQ(
+            escapeChar.size(), 1, "Escape string must be a single character");
+        vectorWriter.current() = applyRow(
+            inputReader[row], patternReader[row], escapeChar.data()[0]);
+        vectorWriter.commit(true);
+      });
+    }
+
+    vectorWriter.finish();
+    context.moveOrCopyResult(localResult, rows, result);
+  }
+
+ private:
+  mutable folly::F14FastMap<
+      std::pair<StringView, std::optional<char>>,
```

Comment:
Why StringView? Shouldn't it be std::string? What if backing string goes away?

Reply:
yep thats a bug

Reply:
Big Big bug